From cc5095b17a82e77aee7360c8d488c75011d44d2c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Mon, 29 Jul 2019 11:10:21 +0200
Subject: [PATCH] Import sage.all initiall so that imports resolve correctly

otherwise we get a
```
import: 'surface_dynamics'

Traceback (most recent call last):
  File "/home/conda/feedstock_root/build_artifacts/surface-dynamics_1564390545642/test_tmp/run_test.py", line 2, in <module>
    import surface_dynamics
  File ".../surface_dynamics/__init__.py", line 21, in <module>
    from .topology.all import *
  File ".../surface_dynamics/topology/all.py", line 10, in <module>
    from .fat_graph import FatGraph
  File ".../surface_dynamics/topology/fat_graph.py", line 19, in <module>
    from sage.rings.integer_ring import ZZ
  File "sage/rings/integer.pxd", line 7, in init sage.rings.integer_ring (build/cythonized/sage/rings/integer_ring.c:17024)
  File "sage/rings/rational.pxd", line 8, in init sage.rings.integer (build/cythonized/sage/rings/integer.c:51949)
  File "sage/rings/rational.pyx", line 81, in init sage.rings.rational (build/cythonized/sage/rings/rational.c:39566)
ImportError: cannot import name ZZ
```
---
 surface_dynamics/__init__.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/surface_dynamics/__init__.py b/surface_dynamics/__init__.py
index b49622b..4f72c0e 100644
--- a/surface_dynamics/__init__.py
+++ b/surface_dynamics/__init__.py
@@ -10,6 +10,9 @@ import warnings
 warnings.filterwarnings('default',
     r'\[surface_dynamics].*')
 
+# Make sure that sage's imports are going to resolve in the correct order
+import sage.all
+
 from .topology.all import *
 from .flat_surfaces.all import *
 from .interval_exchanges.all import *
-- 
2.22.0

